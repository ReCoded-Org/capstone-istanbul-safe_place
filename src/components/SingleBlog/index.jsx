import React, { useState, useEffect } from "react";
import "./index.scss";
import { Container, Col, Row } from "react-bootstrap";
import { useParams, Link } from "react-router-dom";
import { ReactComponent as FacebookIcon } from "../../images/icons/facebookIcon.svg";
import { ReactComponent as TwitterIcon } from "../../images/icons/twitterIcon.svg";
import { ReactComponent as LinkedInIcon } from "../../images/icons/linkedInIcon.svg";
import { ReactComponent as InstagramIcon } from "../../images/icons/instagramIcon.svg";
import { useTranslation } from "react-i18next";

const RELATED_POSTS_TO_SHOW = 4;

export default function SingleBlog(props) {
  const { t } = useTranslation();
  const params = useParams();
  const [blogPost, setBlogPost] = useState();

  useEffect(() => {
    const fetchSingleBlogPost = async () => {
      const data = await fetch(
        `https://public-api.wordpress.com/wp/v2/sites/safeplace102505649.wordpress.com/posts/${params.id}`
      );
      const post = await data.json();
      setBlogPost(post);
    };
    fetchSingleBlogPost();
  }, [params]);

  if (!blogPost) {
    return <h2>{t("singleBlog.loadingMessage")}</h2>;
  }
  const relatedPosts = blogPost["jetpack-related-posts"]
    .map((relatedPost) => {
      return (
        <div>
          <Link to={`/blog/${relatedPost.id}`}>
            <img
              className="relatedPostImg"
              alt="Blog related post"
              src={relatedPost.img.src}
            />
          </Link>
          <h6 className="relatedPostTitle">{relatedPost.title}</h6>
        </div>
      );
    })
    .slice(0, RELATED_POSTS_TO_SHOW);

  return (
    <Container>
      <Row className="blogContent">
        <Col xs={12} md={8}>
          <a href="./">
            <h5 className="backToBlogBtn">
              <span>&#8592;</span>
              {t("singleBlog.backToBlogBtn")}
            </h5>
          </a>
          <div
            className="article"
            dangerouslySetInnerHTML={{
              __html: blogPost.content.rendered,
            }}
          ></div>
          <h4>{t("singleBlog.socialMediaIconsTitle")}</h4>
          <div className="socialMediaIcons">
            <FacebookIcon />
            <TwitterIcon />
            <LinkedInIcon />
            <InstagramIcon />
          </div>
        </Col>
        <Col md={4} className="relatedPostsContainer">
          <h5>{t("singleBlog.relatedPostsContainerTitle")}</h5>
          <div className="relatedPosts">{relatedPosts}</div>
        </Col>
      </Row>
    </Container>
  );
}
