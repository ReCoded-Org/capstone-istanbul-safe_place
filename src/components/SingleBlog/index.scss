@import "../../style/variables";
@import "../../style/buttons";

.blogContent {
  font-family: $mainFontFamily;
  color: $darkColor;

  img {
    margin: 1rem 0;
    border-radius: 1.5rem;
    position: relative;
    object-fit: cover;
    object-position: 0 25%;
    width: 100%;
    height: 20rem;
  }

  .relatedPostImg{
    width: 100%;
    height: 25%;
  }

  a:hover {
    text-decoration: underline;
    color: $primaryColor;
  }

  h4 {
    margin-top: 3rem;
    font-weight: bold;
  }

  h5 {
    margin-top: 3rem;
    font-weight: bold;
    color: $primaryColor;
  }

  .socialMediaIcons {
    width: 15rem;
    display: flex;
    justify-content: space-between;
    margin-top: 1rem;
    margin-bottom: 3rem;
    svg {
      width: 1.5rem;
      height: auto;

      path {
        fill: $darkColor;
      }
    }
  }
}
