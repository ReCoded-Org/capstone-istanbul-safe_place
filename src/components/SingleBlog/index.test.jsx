import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { render } from "react-dom";
import { act } from "react-dom/test-utils";
import SingleBlog from "./index";

let container;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

describe("Single blog", () => {
  const MOCK_DATA = {
    content: {
      rendered: "You can find safe place for yourself here",
    },
    "jetpack-related-posts": [
      {
        id: 61,
        title: "Women and health",
        img: {
          src:
            "https://safeplace102505649.files.wordpress.com/2020/11/smilinggirl.jpg?w=350&h=200&crop=1",
        },
      },
    ],
  };

  it("Renders without crashing", async () => {
    jest.spyOn(global, "fetch").mockImplementation(() =>
      Promise.resolve({
        json: () => Promise.resolve(MOCK_DATA),
      })
    );

    await act(async () => {
      render(
        <Router>
          <SingleBlog />
        </Router>,
        container
      );
    });

    expect(container).toMatchSnapshot();
  });
});