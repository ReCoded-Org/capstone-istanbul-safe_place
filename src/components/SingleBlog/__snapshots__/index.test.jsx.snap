// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single blog Renders without crashing 1`] = `
<div>
  <div
    class="container"
    style=""
  >
    <div
      class="blogContent row"
    >
      <div
        class="col-md-8 col-12"
      >
        <a
          href="./"
        >
          <h5
            class="backToBlogBtn"
          >
            <span>
              ‚Üê
            </span>
            singleBlog.backToBlogBtn
          </h5>
        </a>
        <div
          class="article"
        >
          You can find safe place for yourself here
        </div>
        <h4>
          singleBlog.socialMediaIconsTitle
        </h4>
        <div
          class="socialMediaIcons"
        >
          <svg>
            facebookIcon.svg
          </svg>
          <svg>
            twitterIcon.svg
          </svg>
          <svg>
            linkedInIcon.svg
          </svg>
          <svg>
            instagramIcon.svg
          </svg>
        </div>
      </div>
      <div
        class="relatedPostsContainer col-md-4"
      >
        <h5>
          singleBlog.relatedPostsContainerTitle
        </h5>
        <div
          class="relatedPosts"
        >
          <div>
            <a
              href="/blog/61"
            >
              <img
                alt="Blog related post"
                class="relatedPostImg"
                src="https://safeplace102505649.files.wordpress.com/2020/11/smilinggirl.jpg?w=350&h=200&crop=1"
              />
            </a>
            <h6
              class="relatedPostTitle"
            >
              Women and health
            </h6>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
